hugr-py/src/hugr/serialization/ops.py

```diff
@@ -142,14 +136,29 @@ class Config:
                 "A Sum variant For any Sum type where this value meets the type "
                 "of the variant indicated by the tag."
             ),
-            "required": ["parent", "op", "c", "tag", "typ", "vs"],
+            "required": ["c", "tag", "typ", "vs"],
         }
 
 
-class Const(RootModel):
-    """A constant operation."""
+class Value(RootModel):
+    """A constant Value."""
+
+    root: ExtensionValue | FunctionValue | TupleValue | SumValue = Field(
+        discriminator="c"
+    )
+
 
-    root: ExtensionConst | FunctionConst | Tuple | Sum = Field(discriminator="c")
+class Const(BaseOp):
+    """A Const operation definition."""
+
+    op: Literal["Const"] = "Const"
+    v: Value = Field()
+
+    class Config:
+        # Needed to avoid random '\n's in the pydantic description
+        json_schema_extra = {
+            "required": ["op", "parent", "input_extensions", "v"],
```

Comment:
I don't think this should be required
```suggestion
            "required": ["op", "parent", "v"],
```